File: java/src/test/java/lol/hub/codewars/catching_car_mileage_numbers/KataTest.java
package lol.hub.codewars.catching_car_mileage_numbers;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class KataTest {
    @Test
    public void test1() {
        assertEquals(0, Kata.isInteresting(3, new int[]{1337, 256}));
    }

    @Test
    public void test2() {
        assertEquals(1, Kata.isInteresting(1335, new int[]{1337, 256}));
    }

    @Test
    public void test3() {
        assertEquals(1, Kata.isInteresting(1336, new int[]{1337, 256}));
    }

    @Test
    public void test4() {
        assertEquals(2, Kata.isInteresting(1337, new int[]{1337, 256}));
    }

    @Test
    public void test5() {
        assertEquals(0, Kata.isInteresting(3236, new int[]{1337, 256}));
    }

    @Test
    public void test6() {
        assertEquals(0, Kata.isInteresting(11207, new int[]{}));
    }

    @Test
    public void test7() {
        assertEquals(0, Kata.isInteresting(11208, new int[]{}));
    }

    @Test
    public void test8() {
        assertEquals(0, Kata.isInteresting(11208, new int[]{1337, 256}));
    }

    @Test
    public void test9() {
        assertEquals(1, Kata.isInteresting(11209, new int[]{}));
    }

    @Test
    public void test10() {
        assertEquals(1, Kata.isInteresting(11209, new int[]{1337, 256}));
    }

    @Test
    public void test11() {
        assertEquals(1, Kata.isInteresting(11210, new int[]{}));
    }

    @Test
    public void test12() {
        assertEquals(2, Kata.isInteresting(11211, new int[]{}));
    }

    @Test
    public void test13() {
        assertEquals(2, Kata.isInteresting(11211, new int[]{1337, 256}));
    }
}

File: go/count_the_monkeys/kata_test.go
package count_the_monkeys

import "testing"

func assertSame[T string | int](a []T, b []T) bool {
	if a == nil && b == nil {
		return true
	}
	if a == nil || b == nil {
		return false
	}
	if len(a) != len(b) {
		return false
	}
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

func Test1(t *testing.T) {
	if !assertSame(monkeyCount(1), []int{1}) {
		t.Errorf("test failed")
	}
}

func Test2(t *testing.T) {
	if !assertSame(monkeyCount(2), []int{1, 2}) {
		t.Errorf("test failed")
	}
}
func Test3(t *testing.T) {
	if !assertSame(monkeyCount(3), []int{1, 2, 3}) {
		t.Errorf("test failed")
	}
}
func Test4(t *testing.T) {
	if !assertSame(monkeyCount(4), []int{1, 2, 3, 4}) {
		t.Errorf("test failed")
	}
}
func Test5(t *testing.T) {
	if !assertSame(monkeyCount(5), []int{1, 2, 3, 4, 5}) {
		t.Errorf("test failed")
	}
}
func Test6(t *testing.T) {
	if !assertSame(monkeyCount(6), []int{1, 2, 3, 4, 5, 6}) {
		t.Errorf("test failed")
	}
}
func Test7(t *testing.T) {
	if !assertSame(monkeyCount(7), []int{1, 2, 3, 4, 5, 6, 7}) {
		t.Errorf("test failed")
	}
}
func Test8(t *testing.T) {
	if !assertSame(monkeyCount(8), []int{1, 2, 3, 4, 5, 6, 7, 8}) {
		t.Errorf("test failed")
	}
}
func Test9(t *testing.T) {
	if !assertSame(monkeyCount(9), []int{1, 2, 3, 4, 5, 6, 7, 8, 9}) {
		t.Errorf("test failed")
	}
}
func Test10(t *testing.T) {
	if !assertSame(monkeyCount(10), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}) {
		t.Errorf("test failed")
	}
}
